paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java

```diff
@@ -195,8 +214,22 @@ public RecordReader<InternalRow> createReader(Split split) throws IOException {
             if (!(split instanceof StatisticTable.StatisticSplit)) {
                 throw new IllegalArgumentException("Unsupported split: " + split.getClass());
             }
+            StatisticSplit statisticSplit = (StatisticSplit) split;
+            LeafPredicate snapshotIdPredicate = statisticSplit.snapshotIdPredicate;
+            Optional<Statistics> statisticsOptional;
+            if (snapshotIdPredicate != null) {
+                Long snapshotId =
+                        (Long)
+                                snapshotIdPredicate
+                                        .visit(LeafPredicateExtractor.INSTANCE)
+                                        .get("snapshot_id")
```

Comment:
Lots of "snapshot_id" in this class, could you add:
private static final String SNAPSHOT_ID = "snapshot_id";

Reply:
And TABLE_TYPE in this class should be private.